app/controllers/SupplementaryUnitController.scala

```diff
@@ -87,4 +97,64 @@ class SupplementaryUnitController @Inject() (
             } yield Redirect(navigator.nextPage(SupplementaryUnitPage(recordId), mode, updatedAnswers))
         )
     }
+
+  def onPageLoadUpdate(mode: Mode, recordId: String): Action[AnyContent] =
+    (identify andThen getData andThen requireData).async { implicit request =>
+      val userAnswerValue = request.userAnswers.get(SupplementaryUnitUpdatePage(recordId))
+
+      val preparedFormFuture: Future[(Form[String], String)] = userAnswerValue match {
+        case Some(value) =>
+          Future.successful((form.fill(value), ""))
+        case None        =>
+          goodsRecordConnector.getRecord(request.eori, recordId).map { record =>
+            val formValue: String = record.supplementaryUnit match {
+              case Some(value) if value != 0 => value.toString
+              case _                         => ""
+            }
+            val measurementUnit   = record.measurementUnit.getOrElse("")
+            (form.fill(formValue), measurementUnit)
+          }
+      }
+
+      preparedFormFuture
+        .map { case (preparedForm, measurementUnit) =>
+          val onSubmitAction: Call = routes.SupplementaryUnitController.onSubmitUpdate(mode, recordId)
+          Ok(view(preparedForm, mode, recordId, measurementUnit, onSubmitAction))
+        }
+        .recover { case ex: Exception =>
+          logger.error(s"Error occurred while fetching record for recordId: $recordId", ex)
+          Redirect(routes.JourneyRecoveryController.onPageLoad().url)
+        }
+    }
+
+  def onSubmitUpdate(mode: Mode, recordId: String): Action[AnyContent] =
+    (identify andThen getData andThen requireData).async { implicit request =>
+      val onSubmitAction: Call = routes.SupplementaryUnitController.onSubmitUpdate(mode, recordId)
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => {
+            val result = for {
+              value <- ottService.getMeasurementUnit(request, recordId)
```

Comment:
So do we get this from OTT because it might not be saved against the goods record already?
Just trying to figure out how necessary this call is at this stage of the process

Reply:
yeah, I was thinking between either calling goodsconnector or ott - then decided to use OTT just to be safe & sure about the measurement unit value.